packages/slice-machine/src/features/onboarding/OnboardingStepDialog/OnboardingStepDialog.tsx

```diff
@@ -21,26 +22,42 @@ export const OnboardingStepDialog = ({
   onClose,
 }: OnboardingStepDialogProps) => {
   const { toggleStepComplete, isStepComplete } = useOnboardingContext();
+  const [ctaOkText, updateCtaOkText] = useUpdatableState(() => {
+    return isStepComplete(step.id) ? "Undo step" : "Mark as done";
+  });
 
-  const markAsDone = () => {
+  const execIfOpen = (fn: () => void) => () => {
     if (!isOpen) return;
+    fn();
+  };
+
+  const markAsDone = () => {
     toggleStepComplete(step.id);
     onClose();
   };
 
   return (
-    <Dialog open={isOpen} onOpenChange={onClose}>
-      <DialogHeader title="Learn" />
+    <Dialog
+      open={isOpen}
+      onOpenChange={onClose}
+      onAnimationEnd={execIfOpen(updateCtaOkText)}
+      onAnimationStart={execIfOpen(updateCtaOkText)}
```

Comment:
⚠️ These two can be passed the same function without having to rely on the double callback.

Reply:
Refactored: https://github.com/prismicio/slice-machine/pull/1406#discussion_r1691438981